src/sagemaker/serve/builder/transformers_builder.py

```diff
@@ -0,0 +1,276 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Transformers build logic with model builder"""
+from __future__ import absolute_import
+import logging
+from abc import ABC, abstractmethod
+from typing import Type
+from packaging.version import Version
+
+from sagemaker.model import Model
+from sagemaker import image_uris
+from sagemaker.serve.utils.local_hardware import (
+    _get_nb_instance,
+)
+from sagemaker.djl_inference.model import _get_model_config_properties_from_hf
+from sagemaker.huggingface import HuggingFaceModel
+from sagemaker.serve.model_server.multi_model_server.prepare import (
+    _create_dir_structure,
+)
+from sagemaker.serve.utils.predictors import TransformersLocalModePredictor
+from sagemaker.serve.utils.types import ModelServer
+from sagemaker.serve.mode.function_pointers import Mode
+from sagemaker.serve.utils.telemetry_logger import _capture_telemetry
+from sagemaker.base_predictor import PredictorBase
+from sagemaker.huggingface.llm_utils import get_huggingface_model_metadata
+
+logger = logging.getLogger(__name__)
+DEFAULT_TIMEOUT = 1800
+
+
+# pylint: disable=W0108
+class Transformers(ABC):
```

Comment:
nit: Transformers term is overloaded. Can you add comment at top of the class to summarize what this class is intended?

Reply:
Addressed